modules/user_accounts/ajax/rejectUser.php

```diff
@@ -1,122 +1,63 @@
 <?php
 /**
- * Media uploader.
+ * User accounts
  *
- * Handles media upload and update actions received from a front-end ajax call
+ * Handles rejection of pending user accounts by a user that has permissions
  *
- * PHP Version 5
+ * Send DB query to remove user based on userID, checking that the
+ * user sending the request has admin permissions and the account
+ * can be removed (no password hash => has never had an activity on
+ * Loris, and is pending).
+ *
+ * PHP Version 7
  *
  * @category Loris
  * @package  User_Accounts
  * @author   Leo T. <lthomas.mcin@gmail.com>
+ *           Zaliqa Rosli <zaliqa.rosli@mcin.ca>
  * @license  Loris license
  * @link     https://github.com/aces/Loris
  */
 
-    define('NO_IDENTIFIER_SUPPLIED', 1);
-    define('USER_NOT_FOUND', 2);
-    define('INCORRECT_PERMISSION', 3);
-    define('ACCOUNT_ACTIVE', 4);
-
+namespace LORIS\user_accounts;
 
-if (!isset($_POST['identifier'])) {
-    throw new LorisException("No identifier supplied", NO_IDENTIFIER_SUPPLIED);
-    exit(1);
-} else {
-    $identifier = $_POST["identifier"];
-    reject_user($identifier);
-}
-/**
-* Checks that logged in user has user_accounts permissions, which is
-* also included in admin permissions
-*
-* @return boolean true if user has admin permission or user accounts false otherwise
-*/
-function _hasPerm()
-{
-    $user =& User::singleton();
-    if (isset($user)) {
-        if (!$user->hasPermission('user_accounts')) {
-            return false;
-        }
-        return true;
-    }
-    return false;
+// Make sure the permission is checked first to avoid malicious userID scans
+if (!(\User::singleton())->hasPermission('user_accounts')) {
+    header("HTTP/1.1 403 Forbidden");
+    header("Content-Type: text/plain");
+    exit(
+        "You do not have the correct permissions for this 
+         operation (need admin or user accounts)"
```

Comment:
I don't think there's any value in the error message saying which permission is missing, and the admin is always implicit..

Reply:
I think that "You do not have valid permissions for ..." would be better than "correct"